httpcache.go

```diff
@@ -498,6 +506,44 @@ func headerAllCommaSepValues(headers http.Header, name string) []string {
 	return vals
 }
 
+type cachedReadCloser struct {
+	resp     *http.Response
+	origBody io.ReadCloser
+	buffer   *bytes.Buffer
+	cache    Cache
+	key      string
+}
+
+func newCachedReadCloser(resp *http.Response, cache Cache, key string) io.ReadCloser {
```

Comment:
Let's document this.

```
// newCachedReadCloser [... does something.]
// It's the caller's responsibility to call Close when done.
```

At least that's my guess of the contract. If it's not documented, we have to guess. Let's document it so guessing is not neccessary.